Makefile

```
@@ -0,0 +1,73 @@
+DBG ?= 0
+
+ifeq ($(DBG),1)
+GOGCFLAGS ?= -gcflags=all="-N -l"
+endif
+
+VERSION     ?= $(shell git describe --always --abbrev=7)
+REPO_PATH   ?= github.com/openshift/cluster-api-provider-equinix-metal
+LD_FLAGS    ?= -X $(REPO_PATH)/pkg/version.Raw=$(VERSION) -extldflags -static
+
+GO111MODULE = on
+export GO111MODULE
+GOFLAGS ?= -mod=vendor
+export GOFLAGS
+GOPROXY ?=
+export GOPROXY
+
+NO_DOCKER ?= 0
+ifeq ($(NO_DOCKER), 1)
+  DOCKER_CMD =
+  IMAGE_BUILD_CMD = imagebuilder
+  CGO_ENABLED = 1
+else
+  DOCKER_CMD := docker run --rm -e CGO_ENABLED=1 -v "$(PWD)":/go/src/github.com/openshift/cluster-api-provider-equinix-metal:Z -w /go/src/openshift/cluster-api-provider-equinix-metal openshift/origin-release:golang-1.15
+  IMAGE_BUILD_CMD = docker build
+endif
+
+.PHONY: vendor
+vendor:
+	go mod tidy
+	go mod vendor
+	go mod verify
+
+.PHONY: generate
+generate: gogen goimports
+	./hack/verify-diff.sh
+
+gogen:
+	go generate ./pkg/... ./cmd/...
+
+.PHONY: fmt
+fmt: ## Go fmt your code
+	hack/go-fmt.sh .
+
+.PHONY: goimports
+goimports: ## Go fmt your code
+	hack/goimports.sh .
+
+.PHONY: vet
+vet: ## Apply go vet to all go files
+	hack/go-vet.sh ./...
+
+.PHONY: test
+test: ## Run tests
+	@echo -e "\033[32mTesting...\033[0m"
+	$(DOCKER_CMD) $(REPO_PATH)/hack/ci-test.sh
```

the `REPO_PATH` needs a little more to get the ci-test.sh running

```suggestion
	$(DOCKER_CMD) /go/src/$(REPO_PATH)/hack/ci-test.sh
```

---

Adding `/go/src` will maybe fix the tests in CI, but will break anyone running this locally unless they have `$GOPATH=/go`, does using a relative path from the Makefile not work?

---

nvm, i see this is already fixed lol

~i think a relative path _should_ work since the tests set this as the working directory. i'll give it a try~

~just looked at gcp makefile and it does use a relative path. @detiber my bad, the proper thing to do for the tests is _remove_ the `$(REPO_PATH)`.~

~hmm, seems the `unit` directive needs a touch too, to remove the repo path variable.~